src/Microsoft.AspNet.Mvc.Core/ApplicationModels/AuthorizationApplicationModelProvider.cs

```diff
@@ -44,6 +44,11 @@ public void OnProvidersExecuting(ApplicationModelProviderContext context)
                     controllerModel.Filters.Add(new AuthorizeFilter(policy));
                 }
 
+                foreach (var attribute in controllerModel.Attributes.OfType<AllowAnonymousAttribute>())
```

Comment:
Should this be checking the concrete attribute, or should this be checking for any attribute that implements the `Microsoft.AspNet.Authorization.IAllowAnonymous` interface?